codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -282,7 +282,7 @@ private int allocateBytesForTree(Http2Stream parent, int connectionWindow) {
                 int weight = child.weight();
                 double weightRatio = weight / (double) totalWeight;
 
-                int bytesForTree = Math.min(nextConnectionWindow, (int) Math.ceil(connectionWindow * weightRatio));
+                int bytesForTree = Math.min(nextConnectionWindow, Math.max(1, (int) (connectionWindow * weightRatio)));
```

Comment:
@Scottmitch Nothing really violates the spec since it just suggests how the algorithm "should" behave. We could do away with the `max` altogether and if bytes remain we can just iterate again. WDYT?